### PR TITLE
scratch: Add version 3.18.1

### LINKED ISSUE #4948: [Request] Add Scratch Desktop for Windows
Dear all

Anyone, please add a new package for [Scratch Desktop](https://scratch.mit.edu/download).

Hope, it's useful for all of us.

Thanks.

### PR DESCRIPTION
**Pull request by schance995**
- Closes #4948

### CODE REVIEW

File: bucket/scratch.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "version": "3.18.1",
+    "description": "Educational drag-and-drop programming environment",
+    "homepage":  "https://scratch.mit.edu/download",
+    "license":  "CC-BY-SA-2.0",
```
Comment (Ash258) [inline bucket/scratch.json:5]:
```suggestion
    "license": {
        "identifier": "Freeware",
        "url": "https://scratch.mit.edu/info/faq"
    },
```

File: bucket/scratch.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "url":  "https://downloads.scratch.mit.edu/desktop/Scratch%20Setup.exe#dl.7z",
+    "hash":  "152d7f8aa902e2ca48bd0210e3ccb370d39ce648f51fbe8fdba87e1ee410162b",
+
+    "pre_install": [
+        "Expand-7zipArchive \"$dir\\`$PLUGINSDIR\\app-32.7z\" \"$dir\"",
+        "Remove-Item \"$dir\\`$*\", \"$dir\\Uninstall*\" -Recurse"
+    ],
+
```
Comment (Ash258) [inline bucket/scratch.json:13]:
```suggestion
```

File: bucket/scratch.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+    "version": "3.18.1",
+    "description": "Educational drag-and-drop programming environment",
+    "homepage":  "https://scratch.mit.edu/download",
```
Comment (Ash258) [inline bucket/scratch.json:4]:
```suggestion
    "homepage": "https://scratch.mit.edu/about",
```

File: bucket/scratch.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    "license":  "CC-BY-SA-2.0",
+    "url":  "https://downloads.scratch.mit.edu/desktop/Scratch%20Setup.exe#dl.7z",
+    "hash":  "152d7f8aa902e2ca48bd0210e3ccb370d39ce648f51fbe8fdba87e1ee410162b",
+
```
Comment (Ash258) [inline bucket/scratch.json:8]:
```suggestion
    "url": "https://downloads.scratch.mit.edu/desktop/Scratch%20Setup.exe#dl.7z",
    "hash": "152d7f8aa902e2ca48bd0210e3ccb370d39ce648f51fbe8fdba87e1ee410162b",
```

**Review by Ash258 — CHANGES_REQUESTED** [review]
When was latest version released?

File: bucket/scratch.json (hunk 4 above)
Comment (schance995) [reply bucket/scratch.json:8]: > When was latest version released?

Release page is [here](https://github.com/LLK/scratch-desktop/releases).
The gui itself is BSD ([license](https://github.com/LLK/scratch-desktop/blob/develop/LICENSE)).
But sharing projects on the website is covered by the freeware license.

**Comment by schance995** [comment]
> When was latest version released?

Release page is [here](https://github.com/LLK/scratch-desktop/releases).
The gui itself is BSD ([license](https://github.com/LLK/scratch-desktop/blob/develop/LICENSE)).
But sharing projects on the website is covered by the freeware license.

**Comment by Ash258** [comment]
Then why there is no checkver and autoudpate?

How is the source code license related to the final product, which has clearly stated, that it is free for good

A

**Comment by schance995** [comment]
That is a fair point, practically speaking what is on the website covers everything license related.

I added checkver but not autoupdate. The installer does not include a version number, and the Github release is source code only.

**Comment by Ash258** [comment]
So rather have thousands of hash check fail issues and manual updates...

Add autoupdate

**Review by Ash258 — APPROVED** [review]

**Pull request merged**